src/axom/quest/InOutOctree.hpp

```diff
@@ -60,3362 +65,1516 @@ namespace quest
-    }
+template <int DIM>
+double InOutOctree<DIM>::DEFAULT_VERTEX_WELD_THRESHOLD = 1E-9;
 
-    /**
-     * \brief Determine if the two given triangles have a vertex in common
-     *
-     * \param t0 The index of the first triangle
-     * \param t1 The index of the second triangle
-     * \param [out] sharedVert The index of the shared vertex, if it exists
-     * \return true if the two triangles have a vertex
-     * in common (returned in sharedVert), false otherwise
-     */
-    bool haveSharedVertex(TriangleIndex t0,
-                          TriangleIndex t1,
-                          VertexIndex& sharedVert) const
-    {
-      // There are two triangles -- check that they have at least one common
-      // vertex
-      TriVertIndices tvRel0 = triangleVertexIndices(t0);
-      TriVertIndices tvRel1 = triangleVertexIndices(t1);
+template <int DIM>
+double InOutOctree<DIM>::DEFAULT_BOUNDING_BOX_SCALE_FACTOR = 1.000123;
 
-      for(int i = 0; i < NUM_TRI_VERTS; ++i)
-      {
-        if(incidentInVertex(tvRel0, tvRel1[i]))
-        {
-          sharedVert = tvRel1[i];
-          return true;
-        }
-      }
-      return false;
-    }
+namespace
+{
+#ifdef AXOM_DEBUG
+/// \brief Utility function to print the vertex indices of a cell
+inline std::ostream& operator<<(std::ostream& os,
+                                const InOutOctree<3>::CellVertIndices& tvInd)
+{
+  os << fmt::format("[{}]", fmt::join(tvInd, ","));
+  return os;
+}
 
-    /**
-     * \brief Determine if the three given triangles have a vertex in common
-     *
-     * \param t0 The index of the first triangle
-     * \param t1 The index of the second triangle
-     * \param t2 The index of the second triangle
-     * \param [out] sharedVert The index of the shared vertex, if it exists
-     * \return true if the three triangles have a vertex in common
-     *  (returned in sharedVert), false otherwise
-     */
-    bool haveSharedVertex(TriangleIndex t0,
-                          TriangleIndex t1,
-                          TriangleIndex t2,
-                          VertexIndex& sharedVert) const
-    {
-      TriVertIndices t0Verts = triangleVertexIndices(t0);
-      TriVertIndices t1Verts = triangleVertexIndices(t1);
-      TriVertIndices t2Verts = triangleVertexIndices(t2);
+inline std::ostream& operator<<(std::ostream& os,
+                                const InOutOctree<3>::CellIndexSet& tSet)
+{
+  os << fmt::format("[{}]", fmt::join(tSet, ","));
+  return os;
+}
 
-      for(int i = 0; i < NUM_TRI_VERTS; ++i)
-      {
-        // check if a vertex from the third triangle is in the first and second
-        if(incidentInVertex(t0Verts, t2Verts[i]) &&
-           incidentInVertex(t1Verts, t2Verts[i]))
-        {
-          sharedVert = t2Verts[i];
-          return true;
-        }
-      }
+#endif
+}  // namespace
 
-      return false;
-    }
+template <int DIM>
+void InOutOctree<DIM>::generateIndex()
+{
+  using Timer = axom::utilities::Timer;
```

Comment:
Just want to make sure you intend to keep the use of Timer here and elsewhere. Generally I've seen it used only for examples and tests.

Reply:
Thanks @bmhan12 -- I'll leave this in until we integrate caliper since it's been a helpful high-level overview of what's going on.
(see: https://github.com/LLNL/axom/issues/142 )

Note: in several cases, user code for InOutOctree toggles a verbosity flag by modifying the `slicLoggingMsgLevel()`, as in the `containment_driver` example: 
https://github.com/LLNL/axom/blob/664f2b3d6467c729ec40bebccb60d395eb8c0ca5/src/axom/quest/examples/containment_driver.cpp#L399-L404
and
https://github.com/LLNL/axom/blob/664f2b3d6467c729ec40bebccb60d395eb8c0ca5/src/axom/quest/examples/containment_driver.cpp#L524-L527